packages/table/src/table.tsx

```diff
@@ -666,20 +667,28 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
     }
 
     protected validateProps(props: ITableProps & { children: React.ReactNode }) {
-        const WARNING_MESSAGE = "Children of Table must be Columns";
-        React.Children.forEach(props.children, (child: React.ReactElement<any>) => {
+        const { children, numFrozenColumns, numFrozenRows } = props;
+
+        React.Children.forEach(children, (child: React.ReactElement<any>) => {
             // save as a variable so that union type narrowing works
-            const cType = child.type;
+            const childType = child.type;
 
-            if (typeof cType === "string") {
-                console.warn(WARNING_MESSAGE);
+            if (typeof childType === "string") {
+                console.warn(Errors.TABLE_NON_COLUMN_CHILDREN);
             } else {
-                const isColumn = cType.prototype === Column.prototype || Column.prototype.isPrototypeOf(cType);
+                const isColumn = childType.prototype === Column.prototype || Column.prototype.isPrototypeOf(childType);
                 if (!isColumn) {
-                    console.warn(WARNING_MESSAGE);
+                    console.warn(Errors.TABLE_NON_COLUMN_CHILDREN);
                 }
             }
         });
+
+        if (numFrozenColumns != null && (numFrozenColumns < 0 || numFrozenColumns > React.Children.count(children))) {
+            throw new Error(Errors.TABLE_NUM_FROZEN_COLUMNS_BOUND);
+        }
+        if (numFrozenRows != null && (numFrozenRows < 0 || numFrozenRows > props.numRows)) {
+            throw new Error(Errors.TABLE_NUM_FROZEN_ROWS_BOUND);
```

Comment:
like, is it worth complaining to the user when a reasonable alternative path is readily available?

in #981 i replaced most of our validation errors with warnings, or removed them altogether in cases where the component could easily handle an 'invalid' value